kokoro/linux/build.sh

```diff
@@ -98,19 +98,30 @@ mkdir luci-py
 tar xzf luci-py.tar.gz --directory luci-py
 export LUCI_CLIENT_ROOT="$PWD/luci-py/client"
 
-# Prepare Swarming files
-# SWARMING_X20_TEST_DIR enables different Kokoro jobs to use different sets of tests
+
+# Initialize some environment variables, unless they have already been set
+# (e.g. by build-nightly.sh)
+if [ -z "${SWARMING_TIMESTAMP}" ] ; then
```

Comment:
don't you want to set this variable unconditionally so you have the most accurate date? Is there a concurrence problem you're trying to avoid?

Reply:
This is set to the timestamp at the beginning of the build in build-nightly.sh, such that this timestamp is close to the start-time of the nightly build, which makes it easier to correlate which build corresponds to which timestamp.

The code you point out sets the timestamp later in the build.sh script for PRs, as a result this timestamp is approx 1h30 behind the start of the build, because we build AGI in between.

I agree this is all a bit convoluted, I plan to clean-up in follow-up PRs.